### DISCUSSION TITLE
Doorbell appears as programmable Button

### DISCUSSION THREAD

**Discussion by MMky13** [RavenSystem/esp-homekit-devices#1580] [MEPLHAA Scripts]
I'm trying to create a Doorbell accessory but using accessories type 13 the ESP will appear to HomeKit as a programmable Button.
This is my used JSON
`{
   "c":{
      "b":[
         {
            "g":0,
            "t":5
         }
      ]
   },
   "a":[
      {
         "t":13,
         "f0":[
            {
               "g":0,
               "t":0
            }
         ]
      }
   ]
}`
Any suggestions if this is a Setup problem or if the problem comes form somewhere else?

**Comment by RavenSystem** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Stateless-button

**Comment by xIlSantox** [comment]
First of all thanks indeed for the great job!
I’m having the same issue. Trying to solve with the configuration switch+contact sensor. Unfortunately once the intercom rings I have the notification both when the sensor opens and closes.
Is there any possibility to get only with the open contact?
My configuration is Shelly 1 V3, with a physical button on gpio 4 and a finder relay connected with gpio 3.
Here the json:
{"c":{"n":"Intercom","tz":"UTC+1,M3.4.0/2,M10.5.0/2"},"a":[{"t":1,"i":0.5,"s":0,"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1,"i":0.5}]},"b":[{"g":4,"p":0},{"g":4,"t":0}]},{"t":5,"i":0.1,"f0":[{"g":3,"t":0}],"f1":[{"g":3}]}]}

Thanks in advance to anyone with a suggestion

**Reply by xrust83** [reply]
"f0":[{"g":3,"t":0}] 
https://github.com/RavenSystem/esp-homekit-devices/wiki/Stateless-button#example
single press ("t": 1), double press ("t": 2) & long press ("t": 3).  
("t":0) - ?
You don't get an error because such an action is judged, but it is used for fixed switches
("t":0) - Single activation. Falling edge

**Reply by xrust83** [reply]
Gpio 4 input  "b":[{"g":4,"p":0},{"g":4,"t":0}]}  + Output "0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1,"i":0.5}]} ?

**Reply by xIlSantox** [reply]
This is something I used for a switch+motion sensor.
I recon the above mentioned json works fine: it does the job. Someone rings downstairs, the finder 42.54 12V ac gives the impulse to the Shelly and I’m able to open via gpio 4.
I’m annoyed because I’ve the notification for both the actions performed by the “sensor” (open/close).
Also annoying the text message saying “intercom open” instead “intercom ringing”

**Reply by xrust83** [reply]
Disable lock notifications in HomKit home settings

**Reply by xIlSantox** [reply]
I’d like to keep them, but only when the contact opens. I was thinking about the option "m", even though I don’t understand whether it should work the way I want. E.g.: {"t":5,"i":0.1,**"m":3**,"f0":[{"g":3,"t":0}],"f1":[{"g":3}]}]}
Any suggestion?

**Comment by xIlSantox** [comment]
Hi everybody, I'm back with some issues.
After updating to HAA V12 I'm unable to use the above mention configuration (contact sensor +  switch).
Before the script changing this was working pretty fine:
`{"c":{"n":"Citofono","tz":"UTC+1,M3.4.0/2,M10.5.0/2"},"a":[{"t":1,"i":0.5,"s":0,"0":{"r":[{"g":4}]},"1":{"r":[{"g":4,"v":1,"i":0.5}]},"b":[{"g":4,"p":0},{"g":4,"t":0}]},{"t":5,"i":0.1,"f0":[{"g":3,"t":0}],"f1":[{"g":3}]}]}
`
The big question is: please, could you – people of good will – help me with the V12 Merlin translation?
Thanks indeed in advance

**Reply by xIlSantox** [reply]
Forgot to mention my configuration:
Shelly 1 V3, with a physical button on gpio 4 and a finder relay connected with gpio 3.

**Comment by xIlSantox** [comment]
Anybody could help please? @RavenSystem I know it could be boring for you, but I’m stuck with the not working code.
Thanks indeed for your patience 🙏🏾

**Reply by RavenSystem** [reply]
`{"c":{"io":[[[4],2],[[5],6,0,1],[[3],6,1]],"n":"Citofono","tz":"UTC+1,M3.4.0/2,M10.5.0/2"},"a":[{"t":1,"i":3,"s":0,"0":{"r":[[4]]},"1":{"r":[[4,1,0.5]]},"b":[[5],[5,0]]},{"t":5,"i":1,"f0":[[3,0]],"f1":[[3]]}]}`

**Comment by xIlSantox** [comment]
Hi José,

You’re the best!
Of course it works 🙌🏾

Cheers,
M
